semgrep/semgrep/config_resolver.py

```diff
@@ -82,9 +82,23 @@ def from_config_list(
             except SemgrepError as e:
                 errors.append(e)
 
-        for config in configs:
+        for i, config in enumerate(configs):
             try:
-                config_dict.update(resolve_config(config))
+                # Patch config_id to fix https://github.com/returntocorp/semgrep/issues/1912
+                resolved_config = resolve_config(config)
```

Comment:
Can we push the fix down to resolve_config level? I think it just makes the key a constant when downloading but we can add some part of the url to the key?

Reply:
Yeah, but we also have to do it for files, dirs, etc.
I do agree that moving it down is cleaner. I'll spend 15 minutes looking at it to see if an easy implementation jumps out; otherwise I'll merge as-is.

Reply:
I gave it another shot pushing the fix down; something about doing so caused the rule_ids not to be output properly...

![image](https://user-images.githubusercontent.com/2374948/98293100-9dd26700-1f62-11eb-84df-982c5376ce3c.png)

Merging as-is.